sui_core/src/authority_client.rs

```diff
@@ -172,3 +174,56 @@ impl AuthorityAPI for AuthorityClient {
         Ok(tr_output)
     }
 }
+
+impl AuthorityClient {
+    /// Handle Batch information requests for this authority.
+    pub async fn handle_batch_streaming_as_stream(
+        &self,
+        request: BatchInfoRequest,
+    ) -> Result<impl Stream<Item = Result<BatchInfoResponseItem, SuiError>>, io::Error> {
+        let tcp_stream = self
+            .0
+            .connect_for_stream(serialize_batch_request(&request))
+            .await?;
+
+        let mut error_count = 0;
+        let TcpDataStream { framed_read, .. } = tcp_stream;
```

Comment:
I have never seen this syntax before, this is useful.

Reply:
yeah if you own something outright (not just & or &mut), you can break it up subject to some visibility constraints I think.